### PR TITLE
Replace signatures in IdentityUpdate

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- https://github.com/xmtp/libxmtp/issues/1033
- Replaces the signatures in `IdentityUpdate` struct to use the new `VerifiedSignature`
- For clients that want to control signature verification, we will need a `MultiChainSmartContractWalletVerifier` that uses different RPC endpoints for each supported `chain_id`
- We can probably refactor the `SignatureRequest` to contain a reference to the verifier, so that it doesn't have to be passed in as part of `add_signature` in the bindings.

## Does this mean SCW signatures work

Not yet. There are still some big TODO items in here, which will cause smart contract wallet signature verification to fail (which it also is in `main`). We don't really want to use the `RpcSmartContractWalletVerifier` in normal clients. Instead we will call a (still unbuilt) RPC method to verify these signatures remotely.

### CODE REVIEW

**Comment by neekolas** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/1051?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1051** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/1051?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1050** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/1050?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1049** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/1049?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1048** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/1048?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1047** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/1047?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: bindings_node/src/mls_client.rs [hunk 1]
```diff
@@ -176,7 +171,8 @@ impl NapiClient {
     signature_bytes: Uint8Array,
     chain_id: BigInt,
     account_address: String,
-    chain_rpc_url: String,
+    // TODO:nm remove this
+    _chain_rpc_url: String,
```
Comment (insipx) [inline bindings_node/src/mls_client.rs:175]:
```suggestion
```

File: bindings_node/src/mls_client.rs [hunk 2]
```diff
@@ -176,7 +171,8 @@ impl NapiClient {
     signature_bytes: Uint8Array,
     chain_id: BigInt,
     account_address: String,
-    chain_rpc_url: String,
+    // TODO:nm remove this
```
Comment (insipx) [inline bindings_node/src/mls_client.rs:174]:
```suggestion
```

File: bindings_wasm/src/mls_client.rs [hunk 3]
```diff
@@ -172,7 +173,8 @@ impl WasmClient {
     signature_bytes: Uint8Array,
     chain_id: u64,
     account_address: String,
-    chain_rpc_url: String,
+    // TODO:nm Remove this
```
Comment (insipx) [inline bindings_wasm/src/mls_client.rs:176]:
```suggestion
```

File: bindings_wasm/src/mls_client.rs [hunk 4]
```diff
@@ -172,7 +173,8 @@ impl WasmClient {
...
+    _chain_rpc_url: String,
```
Comment (insipx) [inline bindings_wasm/src/mls_client.rs:177]:
```suggestion
```

File: bindings_wasm/src/mls_client.rs (hunk 3 above)
Comment (insipx) [reply bindings_wasm/src/mls_client.rs:176]: not sure if they're removed in the next PR

**Review by insipx — APPROVED** [review]

File: bindings_wasm/src/mls_client.rs (hunk 3 above)
Comment (neekolas) [reply bindings_wasm/src/mls_client.rs:176]: I guess I could remove this one now since it has no users. The rest I'm waiting until we are _sure_ about the required inputs, so I don't change the public API twice for no good reason.
